scripts/dts/python-devicetree/src/devicetree/edtlib.py

```diff
@@ -1394,149 +1401,147 @@ def _init_props(self, default_prop_types: bool = False,
                 if name not in _DEFAULT_PROP_SPECS:
                     continue
                 prop_spec = _DEFAULT_PROP_SPECS[name]
-                val = self._prop_val(name, prop_spec.type, False, False, None,
-                                     None, err_on_deprecated)
-                self.props[name] = Property(prop_spec, val, self)
+                prop = self._node.props.get(name)
+                val = self._prop_val(prop_spec, prop, err_on_deprecated)
+                self.props[name] = Property(name, prop_spec, val, self)
 
     def _init_prop(self, prop_spec: PropertySpec,
                    err_on_deprecated: bool) -> None:
         # _init_props() helper for initializing a single property.
         # 'prop_spec' is a PropertySpec object from the node's binding.
 
         name = prop_spec.name
-        prop_type = prop_spec.type
-        if not prop_type:
+
+        if not prop_spec.type:
             _err(f"'{name}' in {self.binding_path} lacks 'type'")
 
-        val = self._prop_val(name, prop_type, prop_spec.deprecated,
-                             prop_spec.required, prop_spec.default,
-                             prop_spec.specifier_space, err_on_deprecated)
+        if prop_spec.pattern and prop_spec.default is not None:
+            _err(f"pattern property '{name}' in {self.binding_path} "
+                  "contains default value")
 
-        if val is None:
-            # 'required: false' property that wasn't there, or a property type
-            # for which we store no data.
-            return
+        matching_props = {}
+        if not prop_spec.pattern:
+            matching_props[name] = self._node.props.get(name)
+        else:
+            for name, prop in self._node.props.items():
+                if prop_spec.pattern.match(name):
+                    matching_props[name] = prop
 
-        enum = prop_spec.enum
-        if enum and val not in enum:
-            _err(f"value of property '{name}' on {self.path} in "
-                 f"{self.edt.dts_path} ({val!r}) is not in 'enum' list in "
-                 f"{self.binding_path} ({enum!r})")
-
-        const = prop_spec.const
-        if const is not None and val != const:
-            _err(f"value of property '{name}' on {self.path} in "
-                 f"{self.edt.dts_path} ({val!r}) "
-                 "is different from the 'const' value specified in "
-                 f"{self.binding_path} ({const!r})")
-
-        # Skip properties that start with '#', like '#size-cells', and mapping
-        # properties like 'gpio-map'/'interrupt-map'
-        if name[0] == "#" or name.endswith("-map"):
-            return
+            if prop_spec.required and not matching_props:
+                _err(f"pattern property '{name}' in {self.binding_path} "
```

Comment:
Bug: `name` is overwritten in the for loop above.
```suggestion
                _err(f"pattern property '{prop_spec.name}' in {self.binding_path} "
```

Reply:
fixed